ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/netty/NettyP2PNetwork.java

```diff
@@ -415,7 +423,31 @@ public void checkMaintainedConnectionPeers() {
     }
   }
 
-  private int connectionCount() {
+  @VisibleForTesting
+  void attemptPeerConnections() {
+    final int availablePeerSlots = Math.max(0, maxPeers - connectionCount());
+    if (availablePeerSlots <= 0) {
+      return;
+    }
+
+    final List<DiscoveryPeer> peers =
+        getDiscoveryPeers()
+            .filter(peer -> peer.getStatus() == PeerDiscoveryStatus.BONDED)
+            .filter(peer -> !isConnected(peer) && !isConnecting(peer))
+            .collect(Collectors.toList());
+    Collections.shuffle(peers);
```

Comment:
Paranoia: should we use `Collections.shuffle(List<?>, Random)` with a secure random?